### PR TITLE
feat: Notification for ongoing call (AR-1236)

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1236" title="AR-1236" target="_blank">StoryAR-1236</a>  Display notification for ongoing calls if the app is in the background
  </td></table>
  <br />
 

# What's new in this PR?

Display notification if app goes to the background while call is ongoing

### Attachments (Optional)

https://user-images.githubusercontent.com/6539347/186634405-f33bee10-3a3c-4c77-a919-9c4c92ca4e4a.mp4

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-896/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-896/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4245864)(4245864)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4245864)(4245864)-dev-release.apk)

**Review by alexandreferris — APPROVED** [review]
LGTM!

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-896/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4246698)(4246698)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4246698)(4246698)-dev-release.apk)

**Review by ohassine — APPROVED** [review]
Just tried it, looks amazing ✨

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt [hunk 1]
```diff
@@ -277,6 +280,48 @@ class WireNotificationManager @Inject constructor(
             }
     }
 
+    /**
+     * Infinitely listen for the screen that use is in
+     * and show notification for OngoingCall if user hides the app from the OngoingCallScreen.
+     * @param userIdFlow Flow of QualifiedID of User
+     * @param currentScreenState StateFlow that informs which screen is currently visible,
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt:287]:
```suggestion
     * Infinitely listen for changes on the screen that user is currently in and show notification for OngoingCall if user hides the app from the OngoingCallScreen.
     * @param userIdFlow Flow of QualifiedID of User
     * @param currentScreenState StateFlow that informs which screen is currently visible,
```

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt [hunk 2]
```diff
@@ -277,6 +280,48 @@ class WireNotificationManager @Inject constructor(
...
+                    // user was in OngoingCallScreen and hides the app,
+                    // so we need to display OngoingCall notifications
+                    coreLogic.getSessionScope(userId)
+                        .calls
+                        .establishedCall()
+                        .map { calls ->
+                            // looking for the call that user was in
+                            val currentOngoingCall = calls.firstOrNull()
+
+                            // combining founded ongoingCall with a userId for displaying a notification
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt:306]:
```suggestion
                            // combine current ongoingCall with a userId in order to display a notification
```

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt [hunk 3]
```diff
@@ -277,6 +280,48 @@ class WireNotificationManager @Inject constructor(
...
+                            Optional.ofNullable(currentOngoingCall) to userId
+                        }
+                } else {
+                    // do nothing if app is not in Background, or there is no user
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt:310]: Should there be any incoming call if there is no user registered? 🤔

**Review by gongracr — APPROVED** [review]
Awesome job Mr. Notifications master 🎓!!

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt (hunk 3 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt:310]: obviously - no, but if user logged out we need to "stop listening" ongoingCalls, it's implemented by `flatmapLatest`

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-896/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4253051)(4253051)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4253051)(4253051)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-896/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4253255)(4253255)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4253255)(4253255)-dev-release.apk)

**Pull request merged**